### ISSUE TITLE
Document and template undocumented policies (neural_dqn, reinforce, lstm)

### ISSUE THREAD

**Issue by espenhk** [espenhk/gamer-ai#55]
## Summary

Three `policy_type` values are wired into `main.py` but absent from the CLAUDE.md policy table, and none of them has a matching grid-search template:

- `neural_dqn` → `NeuralDQNPolicy` (DQN with experience replay + target network, issue #20)
- `reinforce`  → `REINFORCEPolicy` (Monte Carlo policy gradient, issue #21)
- `lstm`       → `LSTMEvolutionPolicy` (recurrent policy trained by evolution, issue #22)

All three are dispatched from `main.py:169-180`:

```python
extra_policy_types = {
    "neural_dqn": _make_neural_dqn,
    "cmaes":      _make_cmaes,
    "reinforce":  _make_reinforce,
    "lstm":       _make_lstm,
}
```

but only `cmaes` appears in the CLAUDE.md policy table, so the other three are discoverable only by reading `main.py`. A new contributor looking at CLAUDE.md would assume they don't exist.

## Motivation

- Discoverability: the "Policies" table in CLAUDE.md is treated as the source of truth (same table describes the `policy_type` YAML field).
- Experimentation: every documented policy now has a `config/gs_*.yaml` template you can copy-and-edit. These three are the only ones still requiring you to write a grid config from scratch.
- Consistency: issues #20, #21, #22 (which introduced these policies) are closed, so the implementations are stable enough to document.

### CLAUDE.md policy table

Add three rows:

| `policy_type` | Class | Algorithm |
|---|---|---|
| `neural_dqn` | `NeuralDQNPolicy` | Deep Q-learning with replay buffer + target network |
| `reinforce`  | `REINFORCEPolicy` | Monte Carlo policy gradient (optional running-mean baseline) |
| `lstm`       | `LSTMEvolutionPolicy` | Recurrent (LSTM) policy, trained by evolutionary search over network weights |

Add a short "Policies" subsection for each describing its hyperparams, matching the existing style for `GeneticPolicy` / `CMAESPolicy`.

### Grid templates

Create three new files under `config/`:

- `config/gs_neural_dqn_v1.yaml` — sweep over `learning_rate`, `batch_size`, `target_update_freq`, `epsilon_decay_steps`. Hyperparams come from `_make_neural_dqn` in `main.py:91-104`.
- `config/gs_reinforce_v1.yaml` — sweep over `learning_rate`, `entropy_coeff`, `baseline`, `hidden_sizes`. Hyperparams from `_make_reinforce` in `main.py:124-137`.
- `config/gs_lstm_v1.yaml` — sweep over `hidden_size`, `population_size`, `initial_sigma`. Hyperparams from `_make_lstm` in `main.py:139-167`.

### grid_search.py wiring

Ensure any new hyperparam keys (e.g. `learning_rate`, `entropy_coeff`, `batch_size`, `target_update_freq`, `epsilon_decay_steps`, `hidden_size`) are added to `_ABBREV` (for readable experiment directory names) and to `_POLICY_PARAM_MAP` where they must be forwarded to `policy_params`.

## Acceptance criteria

- [ ] CLAUDE.md policy table lists all six existing policies plus `neural_dqn`, `reinforce`, `lstm`.
- [ ] `config/gs_neural_dqn_v1.yaml`, `config/gs_reinforce_v1.yaml`, `config/gs_lstm_v1.yaml` exist and each runs end-to-end on a single combo without errors.
- [ ] `python grid_search.py config/gs_<name>_v1.yaml --dry-run` (or equivalent) prints one experiment per Cartesian combo with a readable name.
- [ ] No changes to policy implementation code required — this is pure plumbing + docs.

## Files to change

| File | Change |
|------|--------|
| `CLAUDE.md` | Add 3 rows to policy table + 3 short subsections |
| `config/gs_neural_dqn_v1.yaml` | New grid template |
| `config/gs_reinforce_v1.yaml` | New grid template |
| `config/gs_lstm_v1.yaml` | New grid template |
| `grid_search.py` | Add missing keys to `_ABBREV` and `_POLICY_PARAM_MAP` as needed |
